src/System.Private.CoreLib/shared/System/Convert.Base64.Avx2.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+
+namespace System
+{
+    public static partial class Convert
+    {
+        private static unsafe void EncodeBase64Avx(byte* input, int inputLength, byte* output, int outputLength)
+        {
+            // Based on "Base64 encoding with SIMD instructions" article by Wojciech Muła
+            // http://0x80.pl/notesen/2016-01-12-sse-base64-encoding.html
+            // Encode - https://github.com/WojciechMula/base64simd/blob/master/encode/encode.avx2.cpp
+            // Lookup - https://github.com/WojciechMula/base64simd/blob/master/encode/lookup.avx2.cpp (lookup_pshufb_improved)
+
+            byte* outputCurrent = output;
+            int stride = 2 * 4 * 3;
+
+            Vector256<byte> shiftLut = Vector256.Create(
```

Comment:
Might be more efficient if this was a readonly static (at least until we optimize the JIT to make this a JIT-time constant).

Reply:
@tannergooding yeah I did it in some commit but it adds a static constructor to `Convert` (but probably there is already so it's fine).